### PR TITLE
refs 24_01 Create ui to show all detail image and recyclerview to show all comment

### PR DESCRIPTION
**Pull request by anhpham2403**
https://dev.framgia.com/redmine/issues/151610

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -73,6 +73,10 @@
...
             android:screenOrientation="portrait"
             android:theme="@style/AppTheme"/>
+        <activity
+            android:name=".screen.imageDetail.ImageDetailActivity"
+            android:screenOrientation="portrait"
+            android:theme="@style/AppTheme"/>
     </application>
 
-</manifest>
+</manifest>
```
Comment (Tuanlvt) [inline app/src/main/AndroidManifest.xml]: enter end line

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -73,6 +73,10 @@
             android:name=".screen.videoDetail.VideoDetailActivity"
             android:screenOrientation="portrait"
             android:theme="@style/AppTheme"/>
+        <activity
```
Comment (fs-sournary) [inline app/src/main/AndroidManifest.xml]: format code

File: app/src/main/java/com/framgia/englishconversation/screen/imageDetail/ImageAdapter.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    @Override
+    public int getItemCount() {
+        return mMediaModels == null ? 0 : mMediaModels.size();
+    }
+
+    public class ImageViewHolder extends RecyclerView.ViewHolder {
+        private ItemImageDetailBinding mBinding;
+
+        public ImageViewHolder(ItemImageDetailBinding itemView) {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/imageDetail/ImageAdapter.java]: pls rename itemView to binding or itemViewBinding

File: app/src/main/res/layout/activity_image_detail.xml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+                app:layout_constraintLeft_toLeftOf="parent"
+                app:layout_constraintRight_toRightOf="parent"
+                app:layout_constraintTop_toBottomOf="parent"
+                />
+            <View
+                android:id="@+id/line"
+                android:layout_width="match_parent"
+                android:layout_height="1dp"
+                android:layout_marginLeft="5dp"
+                android:layout_marginRight="5dp"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/activity_image_detail.xml]: hardcode

File: app/src/main/java/com/framgia/englishconversation/screen/imageDetail/ImageDetailActivity.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    @Override
+    protected void onStop() {
+        mViewModel.onStop();
+        super.onStop();
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/englishconversation/screen/imageDetail/ImageDetailActivity.java]: base activity đã viết menthod này r

File: app/src/main/java/com/framgia/englishconversation/screen/imageDetail/ImageDetailActivity.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+                finish();
```
Comment (fs-sournary) [inline app/src/main/java/com/framgia/englishconversation/screen/imageDetail/ImageDetailActivity.java]: missing default case

File: app/src/main/res/layout/activity_image_detail.xml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+                app:layout_behavior="@string/appbar_scrolling_view_behavior"
+                app:layout_constraintLeft_toLeftOf="parent"
+                app:layout_constraintRight_toRightOf="parent"
+                app:layout_constraintTop_toBottomOf="parent"
+                />
+            <View
+                android:id="@+id/line"
+                android:layout_width="match_parent"
+                android:layout_height="1dp"
+                android:layout_marginLeft="5dp"
```
Comment (DoanVanToan) [inline app/src/main/res/layout/activity_image_detail.xml]: hardcode

File: app/src/main/java/com/framgia/englishconversation/screen/imageDetail/ImageDetailPresenter.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
+package com.framgia.englishconversation.screen.imageDetail;
+
+/**
+ * Listens to user actions from the UI ({@link ImageDetailActivity}), retrieves the data and updates
+ * the UI as required.
+ */
+final class ImageDetailPresenter implements ImageDetailContract.Presenter {
+    private static final String TAG = ImageDetailPresenter.class.getName();
```
Comment (fs-sournary) [inline app/src/main/java/com/framgia/englishconversation/screen/imageDetail/ImageDetailPresenter.java]: remove tag

File: app/src/main/res/layout/item_image_detail.xml [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    >
+    <data>
+        <variable
+            name="mediaModel"
+            type="com.framgia.englishconversation.data.model.MediaModel"
+            />
+    </data>
+
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/item_image_detail.xml]: remove this line

File: app/src/main/res/layout/activity_image_detail.xml [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+                app:layout_behavior="@string/appbar_scrolling_view_behavior"
+                app:layout_constraintLeft_toLeftOf="parent"
+                app:layout_constraintRight_toRightOf="parent"
+                app:layout_constraintTop_toBottomOf="@+id/line"
+                />
+
+        </android.support.constraint.ConstraintLayout>
+    </android.support.v4.widget.NestedScrollView>
+
+</layout>
```
Comment (fs-sournary) [inline app/src/main/res/layout/activity_image_detail.xml]: format code

File: app/src/main/res/layout/item_image_detail.xml (hunk 9 above)
Comment (DoanVanToan) [inline app/src/main/res/layout/item_image_detail.xml]: remove blank line

File: app/src/main/res/layout/item_image_detail.xml [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+    <ImageView
+        android:id="@+id/image_view"
```
Comment (DoanVanToan) [inline app/src/main/res/layout/item_image_detail.xml]: remove id if unused

File: app/src/main/AndroidManifest.xml [hunk 12]
```diff
@@ -73,6 +73,12 @@
...
+
+        <activity
+            android:name=".screen.imageDetail.ImageDetailActivity"
+            android:screenOrientation="portrait"
+            android:theme="@style/AppTheme"/>
+
     </application>
 
-</manifest>
+</manifest>
```
Comment (tuanlvt-1117) [inline app/src/main/AndroidManifest.xml]: thêm 1 dòng trống phía dưới </manifest>

File: app/src/main/res/layout/activity_image_detail.xml [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+                app:layout_constraintLeft_toLeftOf="parent"
+                app:layout_constraintRight_toRightOf="parent"
+                app:layout_constraintTop_toBottomOf="parent"
+                />
+            <View
+                android:id="@+id/line"
+                android:layout_width="match_parent"
+                android:layout_height="1dp"
+                android:layout_marginEnd="5dp"
+                android:layout_marginStart="5dp"
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/activity_image_detail.xml]: 42-43-44 hardcode

**Review by DoanVanToan — APPROVED** [review]

**Review by fs-sournary — APPROVED** [review]

**Pull request merged**